core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1212,8 +1223,9 @@ class DAGScheduler(
 
               clearCacheLocs()
 
-              if (!shuffleStage.isAvailable) {
-                // Some tasks had failed; let's resubmit this shuffleStage
+              if (!shuffleStage.isAvailable && noActiveTaskSetManager) {
```

Comment:
Hrmm... yes, @squito , we shouldn't go to else branch when the shuffleStage is not available but active `TaskSetManager` exists.